frame/clearing-house/src/lib.rs

```diff
@@ -460,6 +550,69 @@ pub mod pallet {
 			let _ = <Self as ClearingHouse>::create_market(&config)?;
 			Ok(())
 		}
+
+		/// Opens a position in a market
+		///
+		/// # Overview
+		///
+		/// This may result in the following outcomes:
+		/// * Creation of a whole new position in the market, if one didn't already exist
+		/// * Increase in the size of an existing position, if the trade's direction matches the
+		///   existion position's one
+		/// * Decrease in the size of an existing position, if the trade's direction is counter to
+		///   the existion position's one and its magnitude is smaller than the existing postion's
+		///   size
+		/// * Closing of the existing position, if the trade's direction is counter to the existion
+		///   position's one and its magnitude is approximately the existing postion's size
+		/// * Reversing of the existing position, if the trade's direction is counter to the
+		///   existion position's one and its magnitude is greater than the existing postion's size
+		///
+		/// ![](http://www.plantuml.com/plantuml/svg/FOuzgiD030RxTugN0zZgKna2kOUyLhm2hRJeXrm_9aMgZszWOBP8zAmXVpVM9dLGkVptp1bt0CVtUdBssYl8cscIvjfimCF6jC1TwCdGVWSeMYU7b-CWQ4BehEVIhOBWO3ml7c2JTBaCJZPTfw1-2pRIuzeF)
```

Comment:
For what I understood, green arrows are _reads_ and red ones are _writes_, right?
In that case, why `Asset Transfer` has only black arrows?

Reply:
I didn't know how to color then because the accounts in the diagram are just abstractions, not the actual storage items in the balances or tokens pallets. As such, I'd rather leave the coloring for the storage read/writes that are explicit